extensions/amp-form/0.1/test/test-utils.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isFieldDefault} from '../utils';
+
+describes.realWin('amp-form-utils', {}, env => {
+  let doc;
+
+  beforeEach(() => {
+    doc = env.win.document;
+  });
+
+  describe('isFieldDefault', () => {
+    describe('text field', () => {
+      let textField;
+
+      beforeEach(() => {
+        const html = '<input type="text" value="default">';
+        doc.body.insertAdjacentHTML('afterbegin', html);
```

Comment:
Typically we construct the elements with `doc.createElement` and `setAttribute` instead of relying on HTML-setting methods and properties like `innerHTML` and `insertAdjacentHTML`. Let's do the same for all these tests

Reply:
Elements are inserted as HTML so that the `defaultValue` property is generated correctly. I tried the other approach at first, but `defaultValue` would always be empty. I think this is because:

> `defaultValue` returns the default value as originally **specified in the HTML** that created this object.

Reply:
TIL, thanks! Let's add a comment linking/explaining that so future readers have all the information they need

Reply:
Sorry, I was wrong here! 

I experimented with `element.value = ...` which does not work, but `element.setAttribute('value', ...)` actually sets both the DOM the `defaultValue` property correctly. Will do a small PR to not manipulate the HTML directly.